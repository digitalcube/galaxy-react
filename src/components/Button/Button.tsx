import React, { FC } from "react";
import { useMemo } from "react";
import { Button as ReactstrapButton, ButtonProps as ReactstrapButtonProps } from 'reactstrap'

export type ButtonProps = ReactstrapButtonProps

export const Button: FC<ButtonProps> = (props) => {
    const providedClassName = props.className
    const isBlock = props.block || false
    const className = useMemo(() => {
        const names: string[] = ['btn', 'mt-4', 'font-weight-bold']
        if (providedClassName) 
            names.push(providedClassName)
        if (isBlock)
            names.push('btn-block')

        // tmp
        names.push('btn-login')

        return names.join(' ')
    }, [providedClassName, isBlock])

    return (
        <ReactstrapButton
            {...props}
            className={className + 'btn-colored btn-smaller'}
        />
    )
}

export default Button