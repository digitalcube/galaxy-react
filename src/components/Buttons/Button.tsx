import React, { FC } from 'react';
import {
  Button as ReactstrapButton,
  ButtonProps as ReactstrapButtonProps,
} from 'reactstrap';

export type ButtonProps = Omit<ReactstrapButtonProps, 'size'> & {
  fontWeight?: 'bold' | 'normal' | 'light';
  textAlign?: 'left' | 'center' | 'right';
  rounded?: boolean;
  smaller?: boolean;
};

export const Button: FC<ButtonProps> = ({
  fontWeight, rounded, textAlign, smaller,
   ...props}) => {

  const classNames = [props.className, 'btn-colored']
  if (fontWeight) {
    classNames.push(`font-weight-${fontWeight}`)
  }
  if (textAlign) {
    classNames.push(`text-${textAlign}`)
  }
  if (rounded === true) {
    classNames.push('rounded')
  }
  if (smaller) {
    classNames.push('btn-smaller')
  }

  const className = classNames.filter(Boolean).join(' ')

  return <ReactstrapButton {...props} className={className} />;
};
