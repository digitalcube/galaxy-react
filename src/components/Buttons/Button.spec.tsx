import { render } from '@testing-library/react';
import React from 'react';
import { Button } from './Button';
import { Default as StoryButton } from './Button.stories';

describe('Button', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<StoryButton {...(StoryButton.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it.each([
    [{fontWeight: 'bold' as const}, 'font-weight-bold'],
    [{fontWeight: 'normal' as const}, 'font-weight-normal'],
    [{fontWeight: 'light' as const}, 'font-weight-light'],
    [{textAlign: 'center' as const}, 'text-center'],
    [{textAlign: 'left' as const}, 'text-left'],
    [{textAlign: 'right' as const}, 'text-right'],
    [{rounded: true}, 'rounded'],
  ])('given %p, should contain %p class name', (props, className) => {
    const { container } = render(<Button {...props} />)
    expect(container.firstChild).toHaveClass(className)
  })
});
