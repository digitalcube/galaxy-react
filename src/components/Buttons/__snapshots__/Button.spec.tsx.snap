// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn-colored btn btn-secondary btn-block"
  >
    Button name
  </button>
</DocumentFragment>
`;
